CMakeLists.txt

```diff
@@ -159,9 +168,15 @@ if(${KALDI_USE_PATCH_NUMBER})
     set(KALDI_VERSION "${KALDI_VERSION}.${KALDI_PATCH_NUMBER}")
 endif()
 
-get_third_party(openfst)
-set(OPENFST_ROOT_DIR ${CMAKE_BINARY_DIR}/openfst)
-include(third_party/openfst_lib_target)
+# get_third_party(openfst)
+# set(OPENFST_ROOT_DIR ${CMAKE_BINARY_DIR}/openfst)
+# include(third_party/openfst_lib_target)
+find_library(OpenFST
+            NAMES fst 
+            PATHS ${CMAKE_SOURCE_DIR}/tools/openfst/lib 
+            REQUIRED)
+include_directories(${CMAKE_SOURCE_DIR}/tools/openfst/include)

```

Comment:
Add, before this line,
```
find_path(OpenFST_INCLUDE_DIRS "fst.h" 
          PATHS "${CMAKE_SOURCE_DIR}/tools/openfst/include"
          REQUIRED)
get_filename_component(OpenFST_LIBRARY_DIR ${OpenFST_LIBRARY} DIRECTORY CACHE)
mark_as_advanced(FORCE OpenFST_INCLUDE_DIRS OpenFST_LIBRARY OpenFST_LIBRARY_DIR)
```
Since `find_library` caches the library path, it makes sense to do the same to the include path. As long as the path is cached, the search will not be performed again.